lib/walker.ts

```diff
@@ -660,39 +656,24 @@ class Walker {
   }
 
   hasPatch(record: FileRecord) {
-    const patch = this.patches[record.file];
-
-    if (!patch) {
-      return;
-    }
-
-    return true;
+    return !!this.patches[record.file];
   }
 
   stepPatch(record: FileRecord) {
-    const patch = this.patches[record.file];
-
-    if (!patch) {
-      return;
-    }
-
+    if (!this.patches[record.file]) return;
     let body = (record.body || '').toString('utf8');
 
-    for (let i = 0; i < patch.length; i += 2) {
-      if (typeof patch[i] === 'object') {
-        if (patch[i].do === 'erase') {
-          body = patch[i + 1];
-        } else if (patch[i].do === 'prepend') {
-          body = patch[i + 1] + body;
-        } else if (patch[i].do === 'append') {
-          body += patch[i + 1];
-        }
-      } else if (typeof patch[i] === 'string') {
-        // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions
-        // function escapeRegExp
-        const esc = patch[i].replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
-        const regexp = new RegExp(esc, 'g');
-        body = body.replace(regexp, patch[i + 1]);
+    for (const patch of this.patches[record.file]) {
+      if (patch.command === 'replace') {
+        // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#escaping
+        const escaped = patch.from.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+        body = body.replace(new RegExp(escaped, 'g'), patch.to);
+      } else if (patch.command === 'append') {
+        body += patch.source;
+      } else if (patch.command === 'erase') {
+        body = patch.source;
+      } else if (patch.command === 'prepend') {
+        body = patch.source + body;
```

Comment:
Wait this is a breaking change... There are users that provide dictionary from package.json, in case them were using the old replace format this would stop working, could you add a fallback to also make replace work as before?